src/couch_index/src/couch_index_server.erl

```diff
@@ -143,8 +144,8 @@ init([N]) ->
         by_db = by_db(N)
     },
     ok = config:listen_for_changes(?MODULE, St),
-    couch_event:link_listener(?MODULE, handle_db_event, St, [all_dbs]),
-    {ok, St}.
+    {ok, Pid} = couch_event:link_listener(?MODULE, handle_db_event, St, [all_dbs]),
```

Comment:
Use `start_event_listener()` here as well and just make it return a Pid not {ok, Pid}? Do the {ok, Pid} match inside  `start_event_listener()`

Reply:
ah whoops

Reply:
I've called start_event_listener but everywhere else (that bothers to keep track) puts the {ok, Pid} at each call site, so I would rather stick with that. It doesn't look too pretty to my eyes to write a function like;

```
foo() ->
   {ok, Pid} = spawn_link(blah),
   Pid.
```